presto-main/src/main/java/com/facebook/presto/operator/aggregation/PrecisionRecallAggregation.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    @CombineFunction
+    public static void combine(
+            @AggregationState PrecisionRecallState state,
+            @AggregationState PrecisionRecallState otherState)
+    {
+        if (state.getTrueWeights() == null && otherState.getTrueWeights() != null) {
+            state.setTrueWeights(new FixedDoubleHistogram(otherState.getTrueWeights()));
+            state.setFalseWeights(new FixedDoubleHistogram(otherState.getFalseWeights()));
+            return;
+        }
+        if (state.getTrueWeights() != null && otherState.getTrueWeights() != null) {
+            state.getTrueWeights().mergeWith(otherState.getTrueWeights());
+            state.getFalseWeights().mergeWith(otherState.getFalseWeights());
+        }
+    }
+
+    protected static class BucketResult
+    {
+        public final Double totalTrueWeight;
+        public final Double totalFalseWeight;
+        public final Double runningTrueWeight;
+        public final Double runningFalseWeight;
+        public final Double left;
+        public final Double right;
+
+        public BucketResult(
+                double left,
+                double right,
+                Double totalTrueWeight,
+                Double totalFalseWeight,
+                double runningTrueWeight,
+                double runningFalseWeight)
+        {
+            this.left = left;
+            this.right = right;
+            this.totalTrueWeight = totalTrueWeight;
+            this.totalFalseWeight = totalFalseWeight;
+            this.runningTrueWeight = runningTrueWeight;
+            this.runningFalseWeight = runningFalseWeight;
+        }
+    }
+
+    protected PrecisionRecallAggregation() {}
+
+    protected static Iterator<BucketResult> getResultsIterator(@AggregationState PrecisionRecallState state)
+    {
+        if (state.getTrueWeights() == null) {
+            return Collections.<BucketResult>emptyList().iterator();
+        }
+
+        final Double totalTrueWeight = Streams.stream(state.getTrueWeights().iterator())
+                .mapToDouble(c -> c.weight)
+                .sum();
+        final Double totalFalseWeight = Streams.stream(state.getFalseWeights().iterator())
+                .mapToDouble(c -> c.weight)
+                .sum();
+
+        return new Iterator<BucketResult>()
+        {
+            Iterator<FixedDoubleHistogram.Bin> trueIt = state.getTrueWeights().iterator();
+            Iterator<FixedDoubleHistogram.Bin> falseIt = state.getFalseWeights().iterator();
+            Double runningFalseWeight = 0.0;
+            Double runningTrueWeight = 0.0;
+
+            @Override
+            public boolean hasNext()
+            {
+                return trueIt.hasNext() && totalTrueWeight > runningTrueWeight;
+            }
+
+            @Override
+            public BucketResult next()
+            {
+                final FixedDoubleHistogram.Bin trueResult = trueIt.next();
```

Comment:
We should throw `NoSuchElementException` if we attempt `next()` but there's no more elements.  It's minor, and although `trueIt` may throw this, it should be thrown here rather than in the inner iterator.